sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample11_MockingClientTypesLiveTests.cs

```diff
@@ -0,0 +1,332 @@
+            {
+                foreach (var eventData in args.EventBatch)
+                {
+                    if (eventData.Body.Length != 0)
+                    {
+                        await bufferedProducerMock.Object.EnqueueEventAsync(eventData);
+                    }
+                }
+            });
+
+            // Create a mock event to fail send
+            var eventToEnqueue = EventHubsModelFactory.EventData(new BinaryData("Sample-Event"));
+            var eventList = new List<EventData>();
+            eventList.Add(eventToEnqueue);
+
+            // Create a set of args to send to the SendEventBatchFailedAsync handler
+            var args = new SendEventBatchFailedEventArgs(eventList, new Exception(), "0", default);
+
+            // Setup EnqueueEventAsync to always pass and return 1
+            bufferedProducerMock.Setup(bp => bp.EnqueueEventAsync(
+                It.IsAny<EventData>(),
+                It.IsAny<CancellationToken>())).ReturnsAsync(1);
+
+            // Set up EnqueueEventsAsync to fail and call the defined fail handler using the
+            // above created args
+            bufferedProducerMock.Setup(bp => bp.EnqueueEventsAsync(
+                It.IsAny<List<EventData>>(),
+                It.IsAny<CancellationToken>())).Callback(() => sendFailed(args));
+
+            var bufferedProducer = bufferedProducerMock.Object;
+
+            bufferedProducer.SendEventBatchFailedAsync += sendFailed;
+
+            #endregion
+        }
+
+        [Test]
+        public async Task MockingEventProcessor()
+        {
+            #region Snippet:EventHubs_Sample11_MockingEventProcessor
+
+            // TestableCustomProcessor is a wrapper class around a CustomProcessor class that exposes
+            // protected methods so that they can be tested
+            var eventProcessorMock =
+                new Mock<TestableCustomProcessor>(5, "consumerGroup", "namespace", "eventHub", Mock.Of<TokenCredential>(), default(EventProcessorOptions))
+                { CallBase = true };
+
+            var testEvents = new[] {
+                EventHubsModelFactory.EventData(new BinaryData("Sample-Event-1")),
+                EventHubsModelFactory.EventData(new BinaryData("Sample-Event-2")),
+                EventHubsModelFactory.EventData(new BinaryData("Sample-Event-3")),
+                EventHubsModelFactory.EventData(new BinaryData("Sample-Event-4")),
+            };
+
+            var eventList = new List<EventData>(testEvents);
+
+            var eventProcessor = eventProcessorMock.Object;
+
+            // Call the wrapper method in order to reach proctected method within a custom processor
+            // Using It.Is allows the test to set the PartitionId value, even though the setter is protected
+            await eventProcessor.TestOnProcessingEventBatchAsync(eventList, It.Is<EventProcessorPartition>(value => value.PartitionId == "0"));
```

Comment:
Wasn't actually sure how someone should test a custom event processor - this was my best guess